apps/src/p5lab/spritelab/commands/criterionCommands.js

```diff
@@ -375,6 +375,60 @@ export const commands = {
     return false;
   },
 
+  // Returns true if a minimum number of sprites are in the specified group.
+  minimumGroupSprites(group, min = 1) {
+    const spriteIds = this.getSpriteIdsInUse();
+    const groupSpriteIds = spriteIds.filter(
+      id => this.nativeSpriteMap[id].group === group
+    );
+    return groupSpriteIds.length >= min;
+  },
+
+  // Returns true if a minimum number of sprites have no group.
+  minimumNonGroupSprites(min = 1) {
+    const spriteIds = this.getSpriteIdsInUse();
+    const groupSpriteIds = spriteIds.filter(
+      id => !this.nativeSpriteMap[id].group
+    );
+    return groupSpriteIds.length >= min;
+  },
+
+  // Returns true if sprites with a minimum number of costumes are in the specified group.
+  minimumCostumesForGroup(group, min = 1) {
+    const uniqueCostumes = [];
+    const spriteIds = this.getSpriteIdsInUse();
+    const spritesInUse = spriteIds.map(id => this.nativeSpriteMap[id]);
+    const groupSprites = spritesInUse.filter(sprite => sprite.group === group);
+
+    for (const sprite of groupSprites) {
+      const costume = sprite.getAnimationLabel();
+      if (!uniqueCostumes.includes(costume)) {
```

Comment:
nit: In the future, could use a set for this, but no need to change.

Reply:
I had that thought but wasn’t which was preferred/more readable.

Reply:
Either is fine! There aren't any performance issues with an array this size, so it really doesn't matter.